src/main/java/org/terasology/accelerationblocks/AccelerationSystem.java

```diff
@@ -0,0 +1,150 @@
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class AccelerationSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+    private static final Logger logger = LoggerFactory.getLogger(AccelerationSystem.class);
+
+    @In
+    private BlockEntityRegistry blockEntityProvider;
+
+    @In
+    private Time time;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private WorldProvider worldProvider;
+
+    @Override
+    public void update(float delta) {
+        for (EntityRef entity : entityManager.getEntitiesWith(AccelerationComponent.class, LocationComponent.class)) {
+            AccelerationComponent acceleration = entity.getComponent(AccelerationComponent.class);
+            LocationComponent location = entity.getComponent(LocationComponent.class);
+
+            applyImpulse(acceleration, entity);
+        }
+    }
+
+    private void applyImpulse(AccelerationComponent acceleration, EntityRef entity) {
+        Vector3f impulse = new Vector3f(acceleration.velocity);
+        entity.send(new CharacterImpulseEvent(impulse));
+    }
+
+    @ReceiveEvent
+    public void onEnterBlock(OnEnterBlockEvent event, EntityRef entity) {
+        //ignores "flying" block
+        if (isAtHeadLevel(event.getCharacterRelativePosition(), entity)) {
+            return;
+        }
+
+
+        Block block = event.getNewBlock();
+        if (blockIsAccelerating(block)) {
+            AccelerationBlockComponent blockAcceleration = block.getEntity().getComponent(AccelerationBlockComponent.class);
+            AccelerationComponent acceleration = entity.getComponent(AccelerationComponent.class);
+
+            if (acceleration == null) {
+                acceleration = new AccelerationComponent();
+                initAcceleration(acceleration, blockAcceleration, block.getDirection());
+                entity.addComponent(acceleration);
+            } else {
+                initAcceleration(acceleration, blockAcceleration, block.getDirection());
+                entity.saveComponent(acceleration);
+            }
+        } else {
+            //check if it was accelerated before and removes acceleration component
+            AccelerationComponent accelerationOld = entity.getComponent(AccelerationComponent.class);
+
+            if (accelerationOld != null) {
+                entity.removeComponent(AccelerationComponent.class);
+            }
+        }
+    }
+
+    private void initAcceleration(AccelerationComponent acceleration, AccelerationBlockComponent blockAcceleration, Side side) {
+        Vector3f velocity = new Vector3f(blockAcceleration.velocity);
+        if(!blockAcceleration.ignoreBlockDirection){
+            Vector3i originalDirection = Side.FRONT.getVector3i();
+            Vector3i blockDirection = side.getVector3i();
+            double angle = Math.acos(originalDirection.getX() * blockDirection.getX() +
+                    originalDirection.getZ() * blockDirection.getZ());
```

Comment:
No need to split this across two lines, 140ish chars is well below our code style recommendation for line length